### PR TITLE
updated view to use react and shared map

### PR DESCRIPTION
**Pull request by micahgodbolt**

### CODE REVIEW

File: examples/hosts/app-integration/external-views/src/app.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    // Unicode 0x2680-0x2685 are the sides of a dice (⚀⚁⚂⚃⚄⚅)
+    const diceChar = String.fromCodePoint(0x267F + value);
+
+    return (
+        <div>
+            <span style={{ fontSize: 50 }}>{diceChar}</span>
+            <button onClick={() => props.data.set(dataKey, Math.floor(Math.random() * 6) + 1)}>Roll</button>
+        </div>
+    );
+};
```
Comment (SamBroner) [inline examples/hosts/app-integration/external-views/src/app.tsx:47]: This looks nice and simple!

**Review by SamBroner — APPROVED** [review]
Looks good.

I don't mind the React semantics, but it's critical we don't over index on them. Is it worth keeping it in vanilla (or picking a framework we're less familiar with?) to force our hand?

Also, NIT, but can we bump the view into another file?

File: examples/hosts/app-integration/external-views/package.json [hunk 2]
```diff
@@ -35,7 +35,12 @@
   "dependencies": {
     "@fluidframework/aqueduct": "^0.31.0",
     "@fluidframework/get-session-storage-container": "^0.31.0",
-    "@fluidframework/get-tinylicious-container": "^0.31.0"
+    "@fluidframework/get-tinylicious-container": "^0.31.0",
```
Comment (ChumpChief) [inline examples/hosts/app-integration/external-views/package.json:38]: All of these changes are in the /external-views sample -- you should instead modify the /external-controller sample

Comment (SamBroner) [reply examples/hosts/app-integration/external-views/package.json:38]: Yikes... I missed this. Thanks @ChumpChief.

File: examples/hosts/app-integration/external-views/src/app.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        throw new Error(`Unable to retrieve data object at URL: "${url}"`);
+    } else if (response.value === undefined) {
+        throw new Error(`Empty response from URL: "${url}"`);
+    }
+
+    return response.value
+};
+
+
+getKeyValueDb().then((db) => {
```
[the remark this answers is not in the document]
Comment (micahgodbolt) [reply examples/hosts/app-integration/external-views/src/app.tsx:70]: my goal was to try and keep anything from create-react-app untouched as possible. React devs are pretty used to React.DOM.render being the entry point to their apps. I know having webpack enter at some start function was really offputting for me for a while.

**Comment by ChumpChief** [comment]
I'd actually recommend leaving it as vanilla JS rather than React here.  We made an explicit choice to use vanilla JS in our Hello World, and I think this example eventually becomes the even easier Hello World.  We should be consistent in our view choice between the two (in fact ideally both use the exact same view because we'll only be changing the model/controller layers between the two).

If we want to switch _both_ over to use React I think it's an orthogonal decision to discuss, and not crucial to this sample's focus.

File: examples/hosts/app-integration/external-views/package.json (hunk 2 above)
Comment (micahgodbolt) [reply examples/hosts/app-integration/external-views/package.json:38]: haha! wrong folder :D

**Pull request closed without merging**